Sage CI job fails because mpfr not found

CC @mkoeppe 

The Sage CI job currently fails:
```
Run (cd build/pkgs/python_flint/src && python -m build --sdist) \
* Creating isolated environment: venv+pip...
* Installing packages in isolated environment:
  - cython
  - meson-python
* Getting build dependencies for sdist...
* Installing packages in isolated environment:
  - ninja >= 1.8.2
* Building sdist...
+ meson setup /home/runner/work/python-flint/python-flint/build/pkgs/python_flint/src /home/runner/work/python-flint/python-flint/build/pkgs/python_flint/src/.mesonpy-6kvl0dcz -Dbuildtype=release -Db_ndebug=if-release -Db_vscrt=md --native-file=/home/runner/work/python-flint/python-flint/build/pkgs/python_flint/src/.mesonpy-6kvl0dcz/meson-python-native-file.ini
The Meson build system
Version: 1.4.1
Source dir: /home/runner/work/python-flint/python-flint/build/pkgs/python_flint/src
Build dir: /home/runner/work/python-flint/python-flint/build/pkgs/python_flint/src/.mesonpy-6kvl0dcz
Build type: native build
Project name: python-flint
Project version: undefined
C compiler for the host machine: cc (gcc 11.4.0 "cc (Ubuntu 11.4.0-1ubuntu1~[22](https://github.com/flintlib/python-flint/actions/runs/9672359498/job/26684533316#step:6:23).04) 11.4.0")
C linker for the host machine: cc ld.bfd 2.38
Cython compiler for the host machine: cython (cython 3.0.10)
Host machine cpu family: x86_64
Host machine cpu: x86_64
Program python found: YES (/tmp/build-env-yc9uu29h/bin/python)
Found pkg-config: YES (/usr/bin/pkg-config) 1.8.0
Run-time dependency python found: YES 3.10
Run-time dependency gmp found: YES 6.2.1
Found CMake: /usr/local/bin/cmake (3.29.6)
WARNING: CMake Toolchain: Failed to determine CMake compilers state
Run-time dependency mpfr found: NO (tried pkgconfig and cmake)

../meson.build:9:11: ERROR: Dependency "mpfr" not found, tried pkgconfig and cmake

A full log can be found at /home/runner/work/python-flint/python-flint/build/pkgs/python_flint/src/.mesonpy-6kvl0dcz/meson-logs/meson-log.txt

ERROR Backend subprocess exited when trying to invoke build_sdist
Error: Process completed with exit code 1.
```
https://github.com/flintlib/python-flint/actions/runs/9672359498/job/26684533316

With meson-python it is apparently necessary to have the build time dependencies available whenmaking an sdist. I'm not sure if there is an existing issue about this. In python-flint CI we just install the dependencies:
https://github.com/flintlib/python-flint/blob/d8c5c8447c83adfa578452fb961a5c5754618e33/.github/workflows/buildwheel.yml#L56-L73

--- a/.github/workflows/ci-sage.yml
+++ b/.github/workflows/ci-sage.yml
@@ -44,7 +44,7 @@ on:
 
 env:
   # Ubuntu packages to install so that the project's "make dist" can succeed
-  DIST_PREREQ:
+  DIST_PREREQ:      libmpfr-dev
   # Name of this project in the Sage distribution
   SPKG:             python_flint
   # Standard setting: Test the current beta release of Sage:

